File: service/src/main/java/by/rekuts/travelagency/service/TourService.java
package by.rekuts.travelagency.service;

import by.rekuts.travelagency.domain.Tour;
import by.rekuts.travelagency.service.impl.CrudService;

/**
 * Interface TourService concerns to Service layer.
 * Realisations of this interface has access to DAO layer
 * in the face of TourRepository interface and could call its methods.
 */
public interface TourService extends CrudService<Tour> {

    /**
     * @param csvFileContent - path to .csv file with tours.
     */
    void importCsvTours(String csvFileContent);
}

File: service/src/main/scala/by/rekuts/travelagency/service/TourService.scala
package by.rekuts.travelagency.service

import by.rekuts.travelagency.domain.Tour

/**
  * Trait TourService concerns to Service layer.
  * Realisations of this trait has access to DAO layer
  * in the face of TourRepository trait and could call its methods.
  */
trait TourService extends CrudService[Tour] {
  /**
    * @param csvFileContent - path to .csv file with tours.
    */
  def importCsvTours(csvFileContent: String): Unit
}
